### PR TITLE
Auto place option

### PR DESCRIPTION
**Pull request by huwshimi**
Add option to auto place unplaced units on commit.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- press shift+?
- tick the checkbox for "Default to automatically place units on commit."
- click save
- drag a charm to the canvas
- click "Commit"
- check that the radio button for "Automatically place" is selected

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2671/
Test FAILed.

File: app/widgets/deployer-bar.js [hunk 1]
```diff
@@ -307,6 +307,7 @@ YUI.add('deployer-bar', function(Y) {
 
       if (container && container.get('parentNode')) {
         container.one('.panel.summary section').setHTML(this.summaryTemplate({
+          autoPlaceDefault: localStorage.getItem('auto-place-default'),
```
Comment (mitechie) [inline app/widgets/deployer-bar.js]: maybe a comment here referencing the hacking config UX since they're a bit disconnected please.

**Comment by mitechie** [comment]
:+1: with one request for a comment tying things together. No qa

**Comment by makyo** [comment]
Appears spurious, rerunning http://ci.jujugui.org:8080/job/juju-gui/2673/

**Comment by bac** [comment]
@huwshimi This branch may need another pass by you based on the comments above and the testing issue.

If you do get it landed on Wednesday could you go ahead and follow the GUI release process (docs/process.rst) to make 1.4.3 of the GUI and bundle it into a new charm?  If so, I'll do the deploy first thing Wednesday morning US/East.  Thanks.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2675/
Test PASSed.

**Comment by huwshimi** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/942

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/943

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/944

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**